primefaces/src/main/resources/META-INF/resources/primefaces/blockui/blockui.js

```diff
@@ -158,6 +158,11 @@ PrimeFaces.widget.BlockUI = PrimeFaces.widget.BaseWidget.extend({
 
         var delay = this.cfg.delay || 0;
         this.timeout = PrimeFaces.queueTask(function() {
+            // #10484: if delayed and AJAX event already finished
+            if ($this.cfg.triggers && PrimeFaces.ajax.Queue.isEmpty()) {
+                PrimeFaces.warn("BlockUI AJAX event completed before showing the block.");
```

Comment:
Warning is ONLY displayed in Development mode not in Production Mode. If you look in your F12 console log you will see lot of debug information but only in Development mode.  It might help explain why they are not seeing their BlockUI if they don't understand why it s not being displayed.